src/lib.rs

```diff
@@ -1 +1,203 @@
+
+        startup_info.cb = size_of::<STARTUPINFOW>() as u32;
+
+        let process_creation_flags = PROCESS_CREATION_FLAGS(0);
+
+        let res = if let Some(directory) = current_directory {
+            let directory = directory.as_os_str();
+            windows::Win32::System::Threading::CreateProcessW(
+                PWSTR::default(),
+                command,
+                std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                inherit_handles,
+                process_creation_flags,
+                std::ptr::null() as *const c_void,
+                directory,
+                &startup_info,
+                &mut process_info as *mut PROCESS_INFORMATION,
+            )
+        } else {
+            windows::Win32::System::Threading::CreateProcessW(
+                PWSTR::default(),
+                command,
+                std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                inherit_handles,
+                process_creation_flags,
+                std::ptr::null() as *const c_void,
+                PWSTR::default(),
+                &startup_info,
+                &mut process_info as *mut PROCESS_INFORMATION,
+            )
+        };
+
+        if res.as_bool() {
+            Ok(Self {
+                process_information: process_info,
+            })
+        } else {
+            Err(Error::CreationFailed(GetLastError().0))
+        }
+    }
+
+    pub fn wait(&self) -> Result<ExitStatus> {
+        let mut exit_code: u32 = 0;
+
+        unsafe {
+            let res = WaitForSingleObject(self.process_information.hProcess, INFINITE);
+
+            if res == WAIT_OBJECT_0 {
+                if GetExitCodeProcess(
+                    self.process_information.hProcess,
+                    &mut exit_code as *mut u32,
+                )
+                .as_bool()
+                {
+                    close_handle(&self.process_information);
+                    Ok(ExitStatus(exit_code))
+                } else {
+                    Err(Error::GetExitCodeFailed(GetLastError().0))
+                }
+            } else {
+                Err(Error::GetExitCodeFailed(GetLastError().0))
+            }
+        }
+    }
+
+    pub fn try_wait(&self) -> Result<Option<ExitStatus>> {
+        let mut exit_code: u32 = 0;
+
+        let res = unsafe {
+            GetExitCodeProcess(
+                self.process_information.hProcess,
+                &mut exit_code as *mut u32,
+            )
+        };
+
+        if res.as_bool() {
+            if exit_code == STATUS_PENDING.0 {
+                Ok(None)
+            } else {
+                close_handle(&self.process_information);
+                Ok(Some(ExitStatus(exit_code)))
+            }
+        } else {
+            Err(Error::GetExitCodeFailed(unsafe { GetLastError().0 }))
+        }
```

Comment:
Same, make the entire thing unsafe instead of localizing it. You are manipulating low-level objects so everything here is unsafe

The function itself doesn't need to be unsafe because you're not letting the user manipulate low level objects (the function has no argument and the return value is safe)